bundle-workflow/src/test_workflow/integ/integ_test_suite.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import subprocess
+
+from git.git_repository import GitRepository
+from paths.script_finder import ScriptFinder
+from test_workflow.local_test_cluster import LocalTestCluster
+
+
+class IntegTestSuite:
+    """
+    Kicks of integration tests for a component based on test configurations provided in
+    test_support_matrix.yml
+    """
+    def __init__(self, component, test_config, bundle_manifest, work_dir):
+        self.component = component
+        self.bundle_manifest = bundle_manifest
+        self.work_dir = work_dir
+        self.test_config = test_config
+        self.script_finder = ScriptFinder()
+        self.repo = GitRepository(self.component.repository, self.component.commit_id, os.path.join(self.work_dir, self.component.name))
+
+    def execute(self):
+        self._fetch_plugin_specific_dependencies()
+        for config in self.test_config.integ_test['test-configs']:
+            self._setup_cluster_and_execute_test_config(config)
+
+    # TODO: fetch pre-built dependencies from s3
+    def _fetch_plugin_specific_dependencies(self):
```

Comment:
This does not scale if new dependencies are added. 
Probably tied to the question above for test_manifest, is the plan to read these dependencies from the test_manifest?

Reply:
this is a stop gap solution. Next we will fetch all maven dependencies from s3 and later on we can replace this with a more sophisticated dependency management solution for plugins

Reply:
Yeah I agree. Simplest thing to do is to pull all the maven artifacts and eventually optimize it.